core/types/receipt.go

```diff
@@ -55,13 +56,20 @@ type Receipt struct {
 	TxHash          common.Hash    `json:"transactionHash" gencodec:"required"`
 	ContractAddress common.Address `json:"contractAddress"`
 	GasUsed         uint64         `json:"gasUsed" gencodec:"required"`
+
+	// Block location fields - not part of the rlp
+	BlockHash        common.Hash `json:"blockHash,omitempty"`
+	BlockNumber      *big.Int    `json:"blockNumber,omitempty"`
+	TransactionIndex *big.Int    `json:"transactionIndex,omitempty"`
```

Comment:
`TransactionIndex` can just be a regular `uint`. Maybe it should be called `BlockIndex` to make it clear that it's the index in the block.

Reply:
I'll change the type 👍 

I got the name `TransactionIndex` from [this RPC spec](https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_gettransactionreceipt). I'm hoping we can continue toward `ethclient` implementing that entire spec.

Reply:
Yeah, it can be called TransactionIndex, it's fine. ethclient will never implement the full RPC API because it is based on core/types and some of the concepts in the web3 RPC API cannot be represented using the types. One example is transaction sender and inclusion info: the RPC API returns inclusion info and sender in the transaction object, but it would be silly to add those fields to `types.Transaction`).

Reply:
I would rather see us separate the interface struct from the storage struct then to compromise on the interface. For example `ethclient` could return something more like what's in `rpcTransaction`. 

It would be good to have a go client that is as functional as rpc clients in other languages and I think `ethclient` should be that interface.

Reply:
I still want to get to the native implementation done. If you're seriously interested in improving Go developer experience using go-ethereum as a library, I'd be super happy if you could help us make that happen by implementing the existing interfaces on the server side